pkg/thanos/statefulset_test.go

```diff
@@ -234,7 +235,7 @@ func TestTracing(t *testing.T) {
 		var containsVolume bool
 		for _, volume := range sset.Spec.Template.Spec.Volumes {
 			if volume.Name == volumeName {
-				if volume.Secret.SecretName == secretName && volume.Secret.Items[0].Key == secretKey {
+				if volume.Secret.SecretName == secretName && volume.Secret.Items[0].Key == secretKey && volume.Secret.Items[0].Path == secretKey {
```

Comment:
shouldn't this be `volume.Secret.Items[0].Path == mountPath`?

Reply:
I don't think so. According to what is in the k8s documentation [1] `Path`  will be the suffix of the value in `mountPath`. So if we take:
```
spec:
  containers:
  - volumeMounts:
    - mountPath: "/etc/foo"
volumes:
  secrets:
    items:
    - key: "bar.yaml"
      path: "bar.yaml"
```

Then the file will be available under `/etc/foo/bar.yaml`, where if we were to use `mountPath` instead we would end up with:


```
spec:
  containers:
  - volumeMounts:
    - mountPath: "/etc/foo"
volumes:
  secrets:
    items:
    - key: "bar.yaml"
      path: "/etc/foo"
```

This would make it so the file would be under `/etc/foo/etc/foo` I think it would also work but it would probably be more confusing

[1] https://kubernetes.io/docs/concepts/configuration/secret/#projection-of-secret-keys-to-specific-paths

Reply:
My bad, sorry for the noise :)